egui/src/widgets/progress_bar.rs

```diff
@@ -0,0 +1,124 @@
+use crate::*;
+
+/// A simple progress bar.
+pub struct ProgressBar {
+    progress: f32,
+    desired_width: Option<f32>,
+    text: String,
+    animate: bool,
+}
+
+impl ProgressBar {
+    pub fn new(progress: f32) -> Self {
+        Self {
+            progress: progress.clamp(0.0, 1.0),
+            desired_width: None,
+            text: "".to_string(),
+            animate: false,
+        }
+    }
+
+    /// The desired width of the bar. Will use all horizonal space if not set.
+    pub fn desired_width(mut self, desired_width: f32) -> Self {
+        self.desired_width = Some(desired_width);
+        self
+    }
+
+    /// A custom text to display on the progress bar.
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn text(mut self, text: impl ToString) -> Self {
+        self.text = text.to_string();
+        self
+    }
+
+    /// Whether to display a loading animation. Note that this require the UI to be redrawn.
+    /// Defaults to `false`.
+    pub fn animate(mut self, animate: bool) -> Self {
+        self.animate = animate;
+        self
+    }
+}
+
+impl Widget for ProgressBar {
+    fn ui(self, ui: &mut Ui) -> Response {
+        let ProgressBar {
+            progress,
+            desired_width,
+            text,
+            mut animate,
+        } = self;
+
+        animate &= progress < 1.0;
+
+        let desired_width = desired_width.unwrap_or(ui.available_size_before_wrap().x);
+        let height = ui.spacing().interact_size.y;
+        let (outer_rect, response) =
+            ui.allocate_exact_size(vec2(desired_width, height), Sense::click());
```

Comment:
```suggestion
            ui.allocate_exact_size(vec2(desired_width, height), Sense::hover());
```

Something that responds to clicks should do on-hover highlights etc.

Clicking progress bars seems like a rare thing to want to do (I know you do it in the widget gallery though), but if we want to support it I think we should do so fully with a `fn sense(mut self, sense: Sense)` in the builder and then using [`Style::interact`](https://docs.rs/egui/0.13.0/egui/style/struct.Style.html#method.interact) to get the right colors on mouse-hover etc.